gtkm/stats/github_fetcher.py

```diff
@@ -1,8 +1,108 @@
+            response = await client.get(URL)
+        return response.text
+    else:
+        cookies = {'gtkm_cookie': cookie}
+        async with httpx.AsyncClient() as client:
+            response = await client.get(URL, cookies=cookies)
+        return response
+
+
+''' Function manage JSON name filed '''
+
+
+def extract_name(JSON_basic_user_data: JSON):
+    if JSON_basic_user_data.get("name") != None:
+        user_name_data = JSON_basic_user_data.get("name")
+
+        if len(user_name_data) == 1:
+            user_name = user_name_data
+            user_surname = None
+        else:
+            user_name = user_name_data.split()[0]
+            user_surname = user_name_data.split()[1]
+    else:
+        user_name = None
+        user_surname = None
+
+    return user_name, user_surname
+
+
+def jsons_parser(basic_user_data: str, repos_info: str):
+    JSON_basic_user_data = JSON.loads(basic_user_data)
+    JSON_repos_info = JSON.loads(repos_info)
+
+    stargaze_count = 0
+    forks_count = 0
+    for element in JSON_repos_info:
+        stargaze_count = stargaze_count + element.get("stargazers_count")
+        forks_count = forks_count + element.get("forks_count")
+
+    user_name, user_surname = extract_name(JSON_basic_user_data)
+    ''' Fetched and parsed user data '''
+    json_summary_file = {
+        'name': user_name,
+        'surname': user_surname,
+        'user_name': JSON_basic_user_data.get("login"),
+        'stargaze_count': stargaze_count,
+        'repos_count': len(JSON_repos_info),
+        'forks_count': forks_count
+    }
+
+    return json_summary_file
+
+
+async def get_user_name(gtkm_cookie):
+    if gtkm_cookie:
+        user_id = await task('http://127.0.0.1:8000/auth/user/id',
+                             cookie=gtkm_cookie)
+
+        user_name = await task("http://127.0.0.1:8000/auth/user/?id=" +
+                               user_id.json()["id"],
+                               cookie=gtkm_cookie)
+
+        return user_name.json()["github_login"]
+
+    return None
+
+
+async def get_basic_info(gtkm_cookie):
+
+    git_user = await get_user_name(gtkm_cookie)
+
+    URL_basic_user_data = URL_BASE + f"/users/{git_user}"
+    basic_user_data = await task(URL_basic_user_data)
+    ''' Check if user exist, if not return error message '''
+    try:
+        JSON_temp_user_check = JSON.loads(basic_user_data)
+        if JSON_temp_user_check.get("message") == "Not Found":
+            return {"Error": "User not found"}
+    except:
+        pass
+
+    URL_repos_info = URL_BASE + f"/users/{git_user}/repos"
+    repos_info = await task(URL_repos_info)
+    ''' Return parsed user and repos info'''
+    return jsons_parser(basic_user_data, repos_info)
+
 
-@github_fetcher.get("/github/stats/general_user")
-def get_general_stats():
-    return {"stars": 1, "github_url": "http://github.com/zdzisiek"}
+@github_fetcher.get("/github/stats/general_user", response_model=BasicUserData)
```

Comment:
![obraz](https://user-images.githubusercontent.com/45544416/166105119-75464796-32fa-4eb4-b90f-7b3fc80e4613.png)

Sądzę, że ten endpoint powinien być nieco lepiej udokumentowany.   
A dokładniej to powinniśmy w jakiś sposób wyjaśnić (najlepiej tak, aby to by ło widoczne w dokumentacji), że użytkownik jest rozpoznawany po ciasteczku.

Reply:
Tak się teraz zastanawiam nad implikacjami zrobienia takiego endpointa.
Czy zrobienie tego w taki sposób nie utrudni nam potem dodanie możliwości udostępniania swojej strony ze statystykami innym użytkownikom?     
Czy nie lepiej zrobić także w tym momencie endpointa w stylu `github/stats/general_user/{user_id}`?   

Co sądzicie @mpfmorawski  @kamilwil ?

Reply:
Myślę, że do MVP nie powinniśmy się tym przejmować. Nad udostępnieniem swoich statystyk innym użytkownikom zajmijmy się później.

Reply:
Nie mówię o tym, aby implementować to teraz, tylko, żeby myśleć o tym, żeby sobie w przyszłości nie skomplikować pracy.   
Poza tym ta zmiana jest minimalna.  
A te API w obecnym stanie jest też trudniejsze do testowania

Reply:
Z mojego punktu widzenia, endpoint `github/stats/general_user/{user_id}` odrobinę nawet uprości kod fetchera. Jednak ta zmiana utrudni trochę pracę frontendu, który będzie musiał pobrać dane z endpointa `auth/user/id`. 

@lapankrz na ostatnim spotkaniu mówił, że kwestie pobrania danych użytkownika można uprościć do tego stopnia, że będziemy musieli wysłać zapytanie tylko do `/auth/user`. Skoro i tak niezbędne jest ciasteczko do weryfikacji, to czy właśnie etapu wyciągania id nie można zrobić już w `/auth/user` (@lapankrz )? To wtedy obecny endpoint fetchera, można zmienić do postaci `github/stats/general_user/{user_name}`, czyli do wersji, która aktualnie jest w dokumentacji.

Reply:
Tak jak mówiłem, jeśli tylko będzie potrzebny taki endpoint, to mogę go dodać

Reply:
Na potrzeby MVP mogę wywołać po swojej stronie oba endpointy osobno, jeśli będzie taka konieczność.